### PR TITLE
miscviewer

### PR DESCRIPTION
**Pull request by gdementen**
misc fixes/code improvements in the viewer

### CODE REVIEW

**Comment by gdementen** [comment]
once this PR is merged and the issue of showing the current file in the viewer is fixed, I think we are good to go for the release

**Comment by alixdamman** [comment]
I would like to fix #181 (custom title) and #256 (put * if unsaved change) before the release. If you agree.

File: larray/viewer.py [hunk 1]
```diff
@@ -2418,7 +2426,8 @@ def open_documentation(self):
 
     def set_current_file(self, filepath):
         self.update_recent_files([filepath])
-        self.currentFile = filepath
+        self.current_file = filepath
+        # TODO: update window title
```
Comment (alixdamman) [inline larray/viewer.py]: or update window title in set_widget_array?

File: larray/viewer.py [hunk 2]
```diff
@@ -2433,17 +2442,17 @@ def update_recent_files(self, filepaths):
...
+
+        # zip will iterate up to the shortest of the two
+        for filepath, action in zip(recent_files, self.recent_file_actions):
+            action.setText(os.path.basename(filepath))
+            action.setStatusTip(filepath)
+            action.setData(filepath)
+            action.setVisible(True)
+        # if we have less recent recent files than actions, hide the remaining actions
+        for action in self.recent_file_actions[len(recent_files):]:
+            action.setVisible(False)
```
Comment (alixdamman) [inline larray/viewer.py]: bad copy/paste from C++ example. Sorry.

File: larray/viewer.py [hunk 3]
```diff
@@ -2176,8 +2176,12 @@ def update_mapping(self, value):
         changed_keys = [k for k in keys_after if value[k] is not self.data.get(k)]
```
Comment (gdementen) [inline larray/viewer.py]: Indeed, that's a good point/catch. But I do not think there is any practical way to do this. We would need to keeps copies of every array in memory just to check them... :(

File: larray/viewer.py (hunk 2 above)
Comment (gdementen) [inline larray/viewer.py]: no worries

File: larray/viewer.py (hunk 1 above)
Comment (gdementen) [inline larray/viewer.py]: this is already the case, but the trick is that set_current_file can be called in cases where set_widget_array is not called (eg on save), and the title should still be update in that case.

**Comment by gdementen** [comment]
I would rather wait for 0.23 to fix #256, because I would like to make the release today and I fear fixing #256 properly will take too much time. Besides it is not urgent. Even Excel has no explicit indicator of modified file. I would like to have it but it is a nice-to-have not a must-have IMO.

**Pull request merged**